### DISCUSSION TITLE
CMAKE build errors

### DISCUSSION THREAD

**Discussion by tashivani** [open-telemetry/opentelemetry-cpp#2434] [Q&A]
Hi All, 

I want to import opentelemetry-cpp library into a c++ project and so I started with building opentelemetry-cpp using CMAKE (version 3.15.2), following instructions provided in this link - https://github.com/open-telemetry/opentelemetry-cpp/blob/main/INSTALL.md#build-instructions-using-cmake. When I run `cmake ..` it throws the errors related to downloading mingw that I'm unsure how to resolve. Can someone please advice on what I need to do to resolve this? I'm new to using cmake and opentelemetry so I would really appreciate any advice you can offer related to this issue or best way to import otel into a c++ project. Thank you!

> cmake ..
> -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
> -- Building for architecture ARCH=x64
> -- OPENTELEMETRY_ABI_VERSION_NO=1
> -- OPENTELEMETRY_VERSION=1.12.0
> -- nlohmann::json dependency satisfied by: git submodule
> -- Using the multi-header code from C:/OpenTelemetry/opentelemetry-cpp/third_party/nlohmann-json/include/
> -- Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR GTEST_MAIN_LIBRARY)
> Installing build tools and dependencies...
> 
> C:\OpenTelemetry\opentelemetry-cpp\build>REM Copyright The OpenTelemetry Authors
> 
> C:\OpenTelemetry\opentelemetry-cpp\build>REM SPDX-License-Identifier: Apache-2.0
> Running with Administrative privilege...
> 'choco' is not recognized as an internal or external command,
> operable program or batch file.
> Visual Studio installations detected:
> 'vswhere' is not recognized as an internal or external command,
> operable program or batch file.
> Building with vs2022 Professional...
> **********************************************************************
> ** Visual Studio 2022 Developer Command Prompt v17.7.1
> ** Copyright (c) 2022 Microsoft Corporation
> **********************************************************************
> [vcvarsall.bat] Environment initialized for: 'x64'
> Using existing vcpkg installation...
> Computing installation plan...
> error: while looking for gtest:x64-windows:
> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\vcpkg\ports\gtest: error: gtest does not exist
> Computing installation plan...
> The following packages will be built and installed:
>     benchmark:x64-windows -> 1.5.3 -- C:\OpenTelemetry\opentelemetry-cpp\tools\ports\benchmark
> Detecting compiler hash for triplet x64-windows...
> Restored 0 package(s) from C:\Users\shivtyagi\AppData\Local\vcpkg\archives in 266 us. Use --debug to see more details.
> Installing 1/1 benchmark:x64-windows...
> Building benchmark:x64-windows...
> -- Installing port from location: C:\OpenTelemetry\opentelemetry-cpp\tools\ports\benchmark
> CMake Warning at scripts/cmake/vcpkg_fail_port_install.cmake:2 (message):
>   vcpkg_fail_port_install has been removed and all values should be moved by
>   adding `supports` field to manifest file or directly adding
>   `benchmark:=fail` to _scripts/ci.baseline.txt_.
> 
>   Please remove `vcpkg_fail_port_install(...)`.
> 
> Call Stack (most recent call first):
>   C:/OpenTelemetry/opentelemetry-cpp/tools/ports/benchmark/portfile.cmake:13 (vcpkg_fail_port_install)
>   scripts/ports.cmake:147 (include)
> 
> 
> -- Note: benchmark only supports static library linkage. Building static library.
> -- Using cached google-benchmark-c05843a9f622db08ad59804c190f98879b76beba.tar.gz.
> -- Cleaning sources at C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/vcpkg/buildtrees/benchmark/src/879b76beba-25008a53f3.clean. Use --editable to skip cleaning for the packages you specify.
> -- Extracting source C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/vcpkg/downloads/google-benchmark-c05843a9f622db08ad59804c190f98879b76beba.tar.gz
> -- Applying patch version.patch
> -- Using source at C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/vcpkg/buildtrees/benchmark/src/879b76beba-25008a53f3.clean
> -- Found external ninja('1.11.0').
> -- Configuring x64-windows
> CMake Warning at scripts/cmake/vcpkg_configure_cmake.cmake:384 (message):
>   The following variables are not used in CMakeLists.txt:
> 
>       CMAKE_DEBUG_POSTFIX
> 
>   Please recheck them and remove the unnecessary options from the
>   `vcpkg_configure_cmake` call.
> 
>   If these options should still be passed for whatever reason, please use the
>   `MAYBE_UNUSED_VARIABLES` argument.
> Call Stack (most recent call first):
>   C:/OpenTelemetry/opentelemetry-cpp/tools/ports/benchmark/portfile.cmake:26 (vcpkg_configure_cmake)
>   scripts/ports.cmake:147 (include)
> 
> 
> -- Building x64-windows-dbg
> -- Building x64-windows-rel
> -- Fixing pkgconfig file: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/vcpkg/packages/benchmark_x64-windows/lib/pkgconfig/benchmark.pc
> -- Using cached mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
> -- Downloading https://mirror.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst -> mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst...
> [DEBUG] To include the environment variables in debug output, pass --debug-env
> [DEBUG] Trying to load bundleconfig from C:\Github\vcpkg\vcpkg-bundle.json
> [DEBUG] Failed to open: C:\Github\vcpkg\vcpkg-bundle.json
> [DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
> [DEBUG] Metrics enabled.
> [DEBUG] Feature flag 'binarycaching' unset
> [DEBUG] Feature flag 'compilertracking' unset
> [DEBUG] Feature flag 'registries' unset
> [DEBUG] Feature flag 'versions' unset
> [DEBUG] Feature flag 'dependencygraph' unset
> Downloading https://mirror.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> error: Failed to download from mirror set
> error: https://mirror.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> error: https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> error: https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> error: https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> error: https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> error: https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> [DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(1048):
> [DEBUG] Time in subprocesses: 0us
> [DEBUG] Time in parsing JSON: 13us
> [DEBUG] Time in JSON reader: 0us
> [DEBUG] Time in filesystem: 4547us
> [DEBUG] Time in loading ports: 0us
> [DEBUG] Exiting after 3.9 s (3842907us)
> 
> CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
> 
>       Failed to download file with error: 1
>       If you are using a proxy, please check your proxy setting. Possible causes are:
> 
>       1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
>          to `https://address:port`. This is not correct, because `https://` prefix
>          claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
>          , etc..) is an HTTP proxy. Try setting `http://address:port` to both
>          HTTP_PROXY and HTTPS_PROXY instead.
> 
>       2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
>          set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
>          The value set by your proxy might be wrong, or have same `https://` prefix issue.
> 
>       3. Your proxy's remote server is out of service.
> 
>       If you've tried directly download the link, and believe this is not a temporary
>       download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
>       to report this upstream download server failure.
> 
> 
> Call Stack (most recent call first):
>   scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
>   scripts/cmake/vcpkg_acquire_msys.cmake:25 (vcpkg_download_distfile)
>   scripts/cmake/vcpkg_acquire_msys.cmake:124 (z_vcpkg_acquire_msys_download_package)
>   scripts/cmake/vcpkg_acquire_msys.cmake:209 (z_vcpkg_acquire_msys_download_packages)
>   scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake:22 (vcpkg_acquire_msys)
>   scripts/cmake/vcpkg_find_acquire_program.cmake:118 (include)
>   scripts/cmake/vcpkg_fixup_pkgconfig.cmake:203 (vcpkg_find_acquire_program)
>   C:/OpenTelemetry/opentelemetry-cpp/tools/ports/benchmark/portfile.cmake:40 (vcpkg_fixup_pkgconfig)
>   scripts/ports.cmake:147 (include)
> 
> 
> error: building benchmark:x64-windows failed with: BUILD_FAILED
> Elapsed time to handle benchmark:x64-windows: 14 s
> Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
> Then check for known issues at:
>   https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+benchmark
> You can submit a new issue at:
>   https://github.com/microsoft/vcpkg/issues/new?title=[benchmark]+Build+error+on+x64-windows&body=Copy+issue+body+from+C%3A%2FProgram%20Files%2FMicrosoft%20Visual%20Studio%2F2022%2FProfessional%2FVC%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
> 
> Computing installation plan...
> The following packages will be built and installed:
>     protobuf:x64-windows -> 3.15.8#2 -- C:\OpenTelemetry\opentelemetry-cpp\tools\ports\protobuf
> Detecting compiler hash for triplet x64-windows...
> Restored 0 package(s) from C:\Users\shivtyagi\AppData\Local\vcpkg\archives in 251 us. Use --debug to see more details.
> Installing 1/1 protobuf:x64-windows...
> Building protobuf:x64-windows...
> -- Installing port from location: C:\OpenTelemetry\opentelemetry-cpp\tools\ports\protobuf
> -- Using cached protocolbuffers-protobuf-436bd7880e458532901c58f4d9d1ea23fa7edd52.tar.gz.
> -- Cleaning sources at C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/vcpkg/buildtrees/protobuf/src/23fa7edd52-47d9944e09.clean. Use --editable to skip cleaning for the packages you specify.
> -- Extracting source C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/vcpkg/downloads/protocolbuffers-protobuf-436bd7880e458532901c58f4d9d1ea23fa7edd52.tar.gz
> -- Applying patch fix-static-build.patch
> -- Applying patch fix-default-proto-file-path.patch
> -- Applying patch port_def.patch
> -- Using source at C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/vcpkg/buildtrees/protobuf/src/23fa7edd52-47d9944e09.clean
> -- Found external ninja('1.11.0').
> -- Configuring x64-windows
> -- Building x64-windows-dbg
> -- Building x64-windows-rel
> -- Fixing pkgconfig file: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/vcpkg/packages/protobuf_x64-windows/lib/pkgconfig/protobuf-lite.pc
> -- Fixing pkgconfig file: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/vcpkg/packages/protobuf_x64-windows/lib/pkgconfig/protobuf.pc
> -- Using cached mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
> -- Downloading https://mirror.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst;https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst -> mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst...
> [DEBUG] To include the environment variables in debug output, pass --debug-env
> [DEBUG] Trying to load bundleconfig from C:\Github\vcpkg\vcpkg-bundle.json
> [DEBUG] Failed to open: C:\Github\vcpkg\vcpkg-bundle.json
> [DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
> [DEBUG] Metrics enabled.
> [DEBUG] Feature flag 'binarycaching' unset
> [DEBUG] Feature flag 'compilertracking' unset
> [DEBUG] Feature flag 'registries' unset
> [DEBUG] Feature flag 'versions' unset
> [DEBUG] Feature flag 'dependencygraph' unset
> Downloading https://mirror.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst
> error: Failed to download from mirror set
> error: https://mirror.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> error: https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> error: https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> error: https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> error: https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> error: https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst: failed: status code 404
> [DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(1048):
> [DEBUG] Time in subprocesses: 0us
> [DEBUG] Time in parsing JSON: 11us
> [DEBUG] Time in JSON reader: 0us
> [DEBUG] Time in filesystem: 4731us
> [DEBUG] Time in loading ports: 0us
> [DEBUG] Exiting after 3.8 s (3736297us)
> 
> CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
> 
>       Failed to download file with error: 1
>       If you are using a proxy, please check your proxy setting. Possible causes are:
> 
>       1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
>          to `https://address:port`. This is not correct, because `https://` prefix
>          claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
>          , etc..) is an HTTP proxy. Try setting `http://address:port` to both
>          HTTP_PROXY and HTTPS_PROXY instead.
> 
>       2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
>          set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
>          The value set by your proxy might be wrong, or have same `https://` prefix issue.
> 
>       3. Your proxy's remote server is out of service.
> 
>       If you've tried directly download the link, and believe this is not a temporary
>       download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
>       to report this upstream download server failure.
> 
> 
> Call Stack (most recent call first):
>   scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
>   scripts/cmake/vcpkg_acquire_msys.cmake:25 (vcpkg_download_distfile)
>   scripts/cmake/vcpkg_acquire_msys.cmake:124 (z_vcpkg_acquire_msys_download_package)
>   scripts/cmake/vcpkg_acquire_msys.cmake:209 (z_vcpkg_acquire_msys_download_packages)
>   scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake:22 (vcpkg_acquire_msys)
>   scripts/cmake/vcpkg_find_acquire_program.cmake:118 (include)
>   scripts/cmake/vcpkg_fixup_pkgconfig.cmake:203 (vcpkg_find_acquire_program)
>   C:/OpenTelemetry/opentelemetry-cpp/tools/ports/protobuf/portfile.cmake:120 (vcpkg_fixup_pkgconfig)
>   scripts/ports.cmake:147 (include)
> 
> 
> error: building protobuf:x64-windows failed with: BUILD_FAILED
> Elapsed time to handle protobuf:x64-windows: 41 s
> Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
> Then check for known issues at:
>   https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+protobuf
> You can submit a new issue at:
>   https://github.com/microsoft/vcpkg/issues/new?title=[protobuf]+Build+error+on+x64-windows&body=Copy+issue+body+from+C%3A%2FProgram%20Files%2FMicrosoft%20Visual%20Studio%2F2022%2FProfessional%2FVC%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
> 
> Computing installation plan...
> error: while looking for ms-gsl:x64-windows:
> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\vcpkg\ports\ms-gsl: error: ms-gsl does not exist
> Computing installation plan...
> error: while looking for nlohmann-json:x64-windows:
> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\vcpkg\ports\nlohmann-json: error: nlohmann-json does not exist
> Computing installation plan...
> error: while looking for abseil:x64-windows:
> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\vcpkg\ports\abseil: error: abseil does not exist
> Computing installation plan...
> error: while looking for grpc:x64-windows:
> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\vcpkg\ports\grpc: error: grpc does not exist
> Computing installation plan...
> error: while looking for prometheus-cpp:x64-windows:
> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\vcpkg\ports\prometheus-cpp: error: prometheus-cpp does not exist
> Computing installation plan...
> error: while looking for curl:x64-windows:
> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\vcpkg\ports\curl: error: curl does not exist
> Computing installation plan...
> error: while looking for thrift:x64-windows:
> C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\vcpkg\ports\thrift: error: thrift does not exist
> Make sure that vcpkg.cmake is set as the CMAKE_TOOLCHAIN_FILE at the START of the cmake build process!
>     Can be command-line arg (cmake -DCMAKE_TOOLCHAIN_FILE=...) or set in your editor of choice.
> CMake Error at C:/Program Files/CMake/share/cmake-3.15/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
>   Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
>   GTEST_MAIN_LIBRARY)
> Call Stack (most recent call first):
>   C:/Program Files/CMake/share/cmake-3.15/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
>   C:/Program Files/CMake/share/cmake-3.15/Modules/FindGTest.cmake:197 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
>   C:/Github/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
>   CMakeLists.txt:606 (find_package)
> 
> 
> -- Configuring incomplete, errors occurred!
> See also "C:/OpenTelemetry/opentelemetry-cpp/build/CMakeFiles/CMakeOutput.log".
> See also "C:/OpenTelemetry/opentelemetry-cpp/build/CMakeFiles/CMakeError.log".

**Comment by tashivani** [comment]
**Quick update:** 
I noticed there is a message in the above error that says

> If you've tried directly download the link, and believe this is not a temporary download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues  to report this upstream download server failure.

And when I click on one of those links like https://mirror.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst, I get a 404 error in the browser so I will try submitting this as an issue on vcpkg and see if that helps.

**Comment by tashivani** [comment]
I got a response back from one of the contributors of vcpkg (link [here](https://github.com/microsoft/vcpkg/issues/35516#issuecomment-1843402590))
They explained that opentelemetry-cpp repo's embedded submodule link is still pointing to an older commit of vcpkg that does not have the newer msys2/mingw patches. And vcpkg uses dozens of the msys2/mingw libraries/tools to build the ports supported [See vcpkg_acquire_msys.cmake](https://github.com/microsoft/vcpkg/blob/b051745c68faa6f65c493371d564c4eb8af34dad/scripts/cmake/vcpkg_acquire_msys.cmake#L262-L639).
Patched with newer mingw-w64-x86_64-libwinpthread-git- [vcpkg_acquire_msys.cmake -patched](https://github.com/microsoft/vcpkg/blob/b051745c68faa6f65c493371d564c4eb8af34dad/scripts/cmake/vcpkg_acquire_msys.cmake#L552-L556).

So, I'll open a ticket with opentelemetry-cpp to update the embedded submodule link and reset the baseline to the versions needed by opentelemetry-cpp

**Comment by ThomsonTan** [comment]
@tashivani , which branch or commit are you using in opentelemetry-cpp repo? The vcpkg repo was updated recently in the repo (see below PR).

https://github.com/open-telemetry/opentelemetry-cpp/pull/2412

**Reply by tashivani** [reply]
I'm on commit d91e5ba
![image](https://github.com/open-telemetry/opentelemetry-cpp/assets/10983792/78fdb938-0945-4979-8495-b53b90da1646)

**Comment by ThomsonTan** [comment]
Is your repor a fresh clone? Or you may need run `git submodule update` to get the updated vcpkg.

**Reply by tashivani** [reply]
It was a fresh clone but I can run git submodule update on it and see if I still get the same error. 

UPDATE: I did a fresh clone today + git submodule update and still see the same error.
